### PR TITLE
Fix PHPStan level 0 problems

### PR DESCRIPTION
**Pull request by szepeviktor**
... and the text of a message.
Please see the diff.

@phpstan shows further problems.
```bash
vendor/bin/phpstan analyze src/ -l 4
```

### CODE REVIEW

File: src/Internal/CosineSimilarity.php [hunk 1]
```diff
@@ -42,8 +42,8 @@ public static function similarity(array $d1, array $d2): float
         $d2Norm = 0;
 
         foreach ($d1 as $i => $xi) {
-            if (isset($v2[$i])) {
-                $prod += $xi * $v2[$i];
+            if (isset($d2[$i])) {
+                $prod += $xi * $d2[$i];
```
Comment (szepeviktor) [inline src/Internal/CosineSimilarity.php:46]: A few line above I read

>      * Where d1 = query document (query terms)
>      * Where d2 = resulting document (matching terms)

You could name `d1` `queryDocument`, and `d2` `resultingDocument` and so on.

File: src/Internal/Filter/Parser.php [hunk 2]
```diff
@@ -127,7 +127,7 @@ private function assertOperator(?Token $token): void
         $type = $token->type ?? null;
 
         if ($type === null || $type < 10 || $type > 30) {
-            $this->syntaxError('valid operator', $token);
+            $this->syntaxError('invalid operator', $token);
```
Comment (szepeviktor) [inline src/Internal/Filter/Parser.php:130]: I see :) the first parameter is called `$expected` - so it is correct.

**Comment by Toflar** [comment]
Hi, thanks for the contribution. So do you actually want to do the work to add phpstan or not? Or what exactly is the purpose of this PR? :)

**Comment by szepeviktor** [comment]
Glad to contribute.
This PR is an inspiration for you. But it includes fixing of two critical errors.

**Comment by Toflar** [comment]
Okay, I know PHPStan but I'm closing this PR then as you don't plan to actually do the work. I don't see the point in having it float around here.

**Comment by szepeviktor** [comment]
I am sorry to open it.

**Pull request closed without merging**